Bwt/BwtProj/Bwt.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace BWT;
+
+/// <summary>
+/// Burrows Wheeler Transformation implementation and inversion of it
+/// </summary>
+public static class Bwt
+{
+    private const int charAmount = 256;
+    
+    /// <summary>
+    /// BWT based on building <see cref="SuffixArray"/> 
+    /// </summary>
+    public static Tuple<char[], int> BwTransformation(String inputString)
+    {
+        var inputLength = inputString.Length;
+        var inputSuffix = new SuffixArray(inputString).sufArray;
+        var bwt = new char[inputLength];
+        for (var i = 0; i < inputLength; i++)
+        {
+            var j = inputSuffix[i] - 1;
+            if (j < 0)
+                j += inputLength;
+            bwt[i] = inputString[j];
+        }
+        return new Tuple<char[], int>(bwt, Array.IndexOf(inputSuffix, 0));
+    }
+    
+    /// <summary>
+    /// Inversion of <see cref="BwTransformation(String)"/> to <see cref="T:char[]"/>
+    /// </summary>
+    public static char[] InverseTransform(Tuple<char[], int> bwt)
+    {
+        
+        var bwtStr = bwt.Item1;
+        var first = bwt.Item2;
+        var next = new int[bwtStr.Length];
+        var count = new int[charAmount + 1];
+        var sortedBwt = new char[bwtStr.Length];
+        foreach (var el in bwtStr)
+            count[el + 1]++;
```

Comment:
```suggestion
        foreach (var el in bwtStr)
        {
            count[el + 1]++;
        }
```
Вообще, по-хорошему, тела всех "составных" операторов заключают в фигурные скобки, даже если там один оператор, но часто народ забивает на это правило в простых случаях. Я предпочитаю всегда фигурные скобки писать --- отлаживал пару раз по три дня баги, когда кто-то забыл поставить скобки и дописал что-то внутрь, не понравилось.

Ну и как обычно, имя el не годится. Электричество? Ёлка?